sdks/go/pkg/beam/core/util/symtab/symtab.go

```diff
@@ -33,14 +32,13 @@ type SymbolTable struct {
 // New creates a new symbol table based on the debug info
 // read from the specified file.
 func New(filename string) (*SymbolTable, error) {
-	b, err := ioutil.ReadFile(filename)
+	f, err := os.Open(filename)
```

Comment:
<!--thread_id:cc_157274930_t; commit:efbc2a94cba7732675576473452f2cbad9533190; resolved:0-->
<!--section:context-quote-->
> **wcn3** wrote:
> The debug libraries take a io.ReaderAt as their argument, taking ownership of that reader. So the file needs to stay open, so closing it here is incorrect.

<!--section:body-->
Ack. We should still close it on the error paths, though.

Reply:
Looking at how debug/* does things, they all make copies of the DWARF data. We're probably in the clear to defer a close at the end of this method.

Reply:
I'm conflicted on this.

I agree we should close it on error paths. Closing it on the main path is a bit sketch. You're correct that the DWARF routine makes a copy of the underlying data, but the elf and macho packages keep a handle to the reader to load auxiliary data which we never use. To close the reader is basically relying on an implementation detail that is subject to change. The costs of keeping an additional file handle open are pretty minor relative to the hassle of getting broken weirdly in the future.

Reply:
+1 to close on error, but not main.